Update api.go

daniron26 opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
Change language code error test from len==2 to 2 <= len <= 5

the current version will reject correct language codes like "fr-CA" and other language codes that has length == 5 

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-plugin-autotranslate/issues/38

daniron26 committed:
Update api.go

Length of target range from en (2) to fr-CA (5)

mattermod commented:
Hello @daniron26,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

Per the [Mattermost Contribution Guide](https://developers.mattermost.com/contribute/getting-started/), we need to add you to the list of [approved contributors](https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true) for the Mattermost project.

**Please help complete the Mattermost [contribution license agreement](http://www.mattermost.org/mattermost-contributor-agreement/)?**
Once you have signed the CLA, please comment with `/check-cla` and confirm that the CLA check is green.

This is a standard procedure for many open source projects.

Please let us know if you have any questions.

We are very happy to have you join our growing community! If you're not yet a member, please consider joining our [Contributors community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors and discuss new opportunities with the core team.

daniron26 commented:
/check-cla

saturninoabril commented:
Awesome, thanks @daniron26! Will take a look at this.

hanzei commented on server/api.go:
```diff
@@ -66,7 +66,7 @@ func (p *Plugin) getGo(w http.ResponseWriter, r *http.Request) {
 
 	target := r.URL.Query().Get("target")
-	if len(target) != 2 {
+	if len(target) < 2 || len(target) > 5 {
```
Are the code that have the length 3 or 4?

saturninoabril replied:
Looking at some typical reference online and currently [supported language by AWS Translate](https://docs.aws.amazon.com/translate/latest/dg/what-is.html#language-pairs), it's either:
- 2 (ex. `fr`), or
- 5 (ex. `fr-CA`), including dash

daniron26 replied:
I don't think there are len 3 or 4 codes, only 2 and 5

hanzei reviewed (changes requested):
Just one question

saturninoabril reviewed (changes requested):
Thanks @daniron26! LGTM. Could you please do the same for [source](https://github.com/mattermost/mattermost-plugin-autotranslate/pull/40/files#diff-3fb3d4184590bc6f747661f3062676031e837b0dd5f2b18640f439be7a2235f6L63)?

daniron26 committed:
Update api.go

updated error checking for source language code for 5 characters codes e.g., fr-CA

daniron26 reviewed (commented):
source language code now allows for 5 character codes like fr-CA

daniron26 reviewed (commented):
CI failed, not sure how to proceed

hanzei commented:
/update-branch

mattermod committed:
Merge branch 'master' into patch-1

hanzei commented on server/api.go:
```diff
@@ -60,13 +60,13 @@ func (p *Plugin) getGo(w http.ResponseWriter, r *http.Request) {
 
 	source := r.URL.Query().Get("source")
-	if len(source) < 2 || len(source) > 4 {
+	if len(source) < 2 || len(source) > 5 {
```
Should we just check for 
```suggestion
	if len(source) != 2 && len(source) != 5 {
```

here and below to be more strict?

daniron26 committed:
Update server/api.go

you're right

Co-authored-by: Ben Schumacher <ben.schumacher@mattermost.com>

hanzei commented on server/api.go:
```diff
@@ -60,13 +60,13 @@ func (p *Plugin) getGo(w http.ResponseWriter, r *http.Request) {
 
 	target := r.URL.Query().Get("target")
-	if len(target) != 2 {
+	if len(target) < 2 || len(target) > 5 {
```
Same here?

saturninoabril commented:
@daniron26 Thanks for updating. Could you please check if the recent change works for you?

daniron26 committed:
Update api.go

Made target language code testing more strict

daniron26 commented:
> Thanks @daniron26! LGTM. Could you please do the same for [source](https://github.com/mattermost/mattermost-plugin-autotranslate/pull/40/files#diff-3fb3d4184590bc6f747661f3062676031e837b0dd5f2b18640f439be7a2235f6L63)?

Its been changed, let me know if I missed anything.

> @daniron26 Thanks for updating. Could you please confirm if the recent change works for you?

Yes it does, happy to be a part of this.

saturninoabril commented:
@daniron26 I'm getting the following error since the `source` could have a value of `auto`:
![Screen Shot 2021-02-16 at 2 44 03 PM](https://user-images.githubusercontent.com/5334504/108028251-7e210380-7066-11eb-835d-399eb4bd3929.png)

With that, we can probably relax the length check in `source`.

daniron26 commented:
> @daniron26 I'm getting the following error since the `source` could have a value of `auto`:
> ![Screen Shot 2021-02-16 at 2 44 03 PM](https://user-images.githubusercontent.com/5334504/108028251-7e210380-7066-11eb-835d-399eb4bd3929.png)
> 
> With that, we can probably relax the length check in `source`.

I think so, changing it back to 2 >= len >= 5

daniron26 committed:
Update api.go

changing back to restricting language code to 2 <= len(source|target) <= 5 for the following cases:
language code = "fr"
language code = "fr-CA"
language code = "auto"

saturninoabril commented on server/api.go:
```diff
@@ -60,13 +60,13 @@ func (p *Plugin) getGo(w http.ResponseWriter, r *http.Request) {
 
 	source := r.URL.Query().Get("source")
-	if len(source) < 2 || len(source) > 4 {
+	if len(source) < 2 || len(source) > 5 { 
```
Failed on lint due to extra whitespace
```suggestion
	if len(source) < 2 || len(source) > 5 {
```

saturninoabril reviewed (approved):
Thanks @daniron26! Tested and passed. LGTM.

saturninoabril committed:
Update server/api.go
